src/TrackedFloat.jl

```diff
@@ -65,30 +61,75 @@ for TrackedFloatN in (:TrackedFloat16, :TrackedFloat32, :TrackedFloat64)
     Base.promote_rule(::Type{Bool},::Type{$TrackedFloatN}) = $TrackedFloatN
   end
 
+  # Use this where an int got wrapped with a TrackedFloat
+  @eval function trunc_if_int(y::$TrackedFloatN)
+    t = trunc(Int, y.val)
+    if abs(y.val - t) < floatmin(y)
+      return t
+    else
+      error("Unable to safely truncate $y into an int")
+    end
+  end
+
+  @eval function Base.ldexp(x::$TrackedFloatN, y::$TrackedFloatN)
+    y_as_int = trunc_if_int(y)
+    (r, injected) = run_or_inject(ldexp, x.val, y_as_int)
+    check_error(ldexp, [x.val, y_as_int], r, injected)
```

Comment:
Oh, `check_error` should als be splatting instead of making a list/vector.

Reply:
Ugh, I wish Julia didn't do this so aggressively. @tcallred do you know of any way to work around the auto-casting of types in a vector?

Just with the placement of this argument it makes it a little more involved to change.

Reply:
Got it: the `convert` function is our friend. We can take a tuple (which we get through the splat) and get a Vector of the most specific common supertype w/o automatic promotion.